app/src/main/java/org/mozilla/fenix/experiments/NimbusSetup.kt

```diff
@@ -88,32 +98,32 @@ fun createNimbus(context: Context, url: String?): NimbusApi {
             // generated code.
             register(observer)
 
-            // This performs the minimal amount of work required to load branch and enrolment data
-            // into memory. If `getExperimentBranch` is called from another thread between here
-            // and the next nimbus disk write (setting `globalUserParticipation` or
-            // `applyPendingExperiments()`) then this has you covered.
-            // This call does its work on the db thread.
-            initialize()
+            val isFirstNimbusRun = context.settings().isFirstNimbusRun
+
+            // We always want `Nimbus.initialize` to happen ASAP and before any features (engine/UI)
+            // have been initialized. For that reason, we use runBlocking here to avoid
+            // inconsistency in the experiments.
+            // We can safely do this because Nimbus does most of it's work on background threads,
+            // except for loading the initial experiments from disk. For this reason, we have a
+            // `joinOrTimeout` to limit the blocking until TIME_OUT_LOADING_EXPERIMENT_FROM_DISK_MS.
+            runBlockingIncrement {
+                val job = initialize(
+                    isFirstNimbusRun || url.isNullOrBlank(),
+                    R.raw.initial_experiments,
+                )
+                // We only read from disk when loading first-run experiments. This is the only time
+                // that we should join and block. Otherwise, we don't want to wait.
+                if (isFirstNimbusRun) {
+                    context.settings().isFirstNimbusRun = false
```

Comment:
We moved the pref update here in case there is some way we put the user into a blocked state and we get an ANR. Their second run will then be safe to continue in a non-blocking way.

cc: @jeddai @jhugman